internal/tcpchecker/compositereporter.go

```diff
@@ -0,0 +1,11 @@
+package tcpchecker
+
+type CompositeReporter struct {
+	Reporters []Reporter
+}
```

Comment:
A small not on this API. If we use `type CompositeReporter []Reporter` usage of the reporter is a bit simpler IMO.

```diff
- reporters := tcpchecker.CompositeReporter{
- 	Reporters: []tcpchecker.Reporter{
- 		&prometheusReporter,
- 	},
- }
+ reporters := tcpchecker.CompositeReporter(&prometheusReporter)
```

Reply:
Yeah let's do that with `...args` 👍